### PR TITLE
Add roadmap to readme

### LINKED ISSUE #1515: high level roadmap for Arrow / Datafusion
@matthewmturner  and I we think it would be valuable to try making a high level roadmap for Arrow / Datafusion.  Specifically, and how this differs from the existing roadmap, we are looking for a rough idea of what people plan on working on with a best efforts guess when they hope to finish.  Of course nothing is binding, the goal is to make visible peoples plans so that the developers as a whole can better organize / plan efforts and users have a rough idea of feature roadmap.

There is a roadmap (https://github.com/apache/arrow-datafusion/blob/master/docs/source/specification/roadmap.md) but im interested in seeing how that translates to a more tangible plan.  im cognizant of not wanting to have something that is too much of a burden to maintain, but some public strategy / structure / plan could be helpful.

Assuming we get enough feedback, we can merge everything in a unified plan.

 started a doc here: https://docs.google.com/document/d/1t64vZwZnXm9MyFj2qz3xcAkSxK3Wu12giS3KrS4nDE0/edit

### PR DESCRIPTION
**Pull request by matthewmturner**
# Which issue does this PR close?

Closes #1515 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by matthewmturner** [comment]
Thank you, @alamb @houqp @xudong963 @yjshen @liukun4515 @hntd187 @realno @pjmore for contributions to roadmap.

Ive created PR here adding roadmap to the datafusion readme.  

Let me know your thoughts :)

File: README.md [hunk 1]
```diff
@@ -141,6 +141,60 @@ datafusion = "6.0.0"
...
+
+### Benchmarking
+
+- Inclusion in Db-Benchmark with all quries covered
+- All TPCH queries covered
+
+### Performance Improvements
+
+- Predicate evaluation
+- Multi-column comparisons that can't be vectorized
```
Comment (xudong963) [inline README.md:163]: `can't be` ?

File: README.md [hunk 2]
```diff
@@ -141,6 +141,60 @@ datafusion = "6.0.0"
...
+
+## Beyond 2022 Q1
+
+There is no clear timeline for the below, but community members have expressed interest in working on these topics.
+
+### DataFusion Core
+
+- Custom SQL support
+- Split DataFusion into multiple crates
+- Push based query execution and code gen
```
Comment (xudong963) [inline README.md:190]:
```suggestions
Push based query execution and codegen
```

**Review by xudong963 — COMMENTED** [review]
Thanks @matthewmturner, nice work!

**Comment by matthewmturner** [comment]
> What is the difference between a new feature and extension? They both sound like kinda the same thing. Semantics aside this sounds good to me.

Extension is specifically referring to topics that would be in `datafusion-contrib` as opposed to a new feature that is in the `datafusion` crate

File: README.md [hunk 3]
```diff
@@ -141,6 +141,60 @@ datafusion = "6.0.0"
...
+
+### DataFusion Core
+
+- Publish official Arrow2 branch
+- Implementation of memory manager (i.e. to enable spilling to disk as needed)
+
+### Benchmarking
+
+- Inclusion in Db-Benchmark with all quries covered
+- All TPCH queries covered
```
[the remark this answers is not in the document]
Comment (xudong963) [reply README.md:158]: I'll check it after getting up

File: README.md (hunk 2 above)
Comment (xudong963) [reply README.md]: both are ok

**Comment by matthewmturner** [comment]
@yahoNanJing FYI - if you want to add anything ballista related just let me know.

File: README.md (hunk 1 above)
Comment (alamb) [reply README.md:163]:
```suggestion
- Improve multi-column comparisons (that can't be vectorized at the moment)
```

**Review by alamb — APPROVED** [review]
Thanks @matthewmturner

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (xudong963) [reply README.md:158]: Sorry, my mistake. I believe the Db-Benchmark you mentioned is `datafusion/benches/`?

[the remark this answers is not in the document]
Comment (matthewmturner) [reply README.md:158]: The db-benchmark im referring to is getting datafusion included here https://h2oai.github.io/db-benchmark/

I've opened a PR there to get it added.

**Comment by alamb** [comment]
Thanks all who contributed to the roadmap -- would love to keep it as a living document

**Comment by matthewmturner** [comment]
@alamb i have a reminder for myself to refresh every 3 months.

File: README.md [hunk 4]
```diff
@@ -141,6 +141,60 @@ datafusion = "6.0.0"
...
 
+# Roadmap
+
+A quarterly roadmap will be published to give the DataFusion community visibility into the priorities of the projects contributors. This roadmap is not binding.
+
+## 2022 Q1
+
+### DataFusion Core
+
+- Publish official Arrow2 branch
```
Comment (tisonkun) [inline README.md:152]: Hi @alamb! I'm curious what "arrow2" means here. Is it related to https://github.com/jorgecarleitao/arrow2?

Comment (matthewmturner) [reply README.md:152]: Yes that's what it refers to

Comment (alamb) [reply README.md:152]: yes, that is exactly what arrow2 means

There is a branch https://github.com/apache/arrow-datafusion/tree/arrow2 and a discussion ticket https://github.com/apache/arrow-datafusion/issues/1532 that has more information if you are interested

Comment (tisonkun) [reply README.md:152]: Thank you!

**Pull request merged**